### ISSUE TITLE
istioctl version 1.13.2 showing invalid error for istioctl x describe pod

### ISSUE THREAD

**Issue by soumyaparui** [istio/istio#38636]
### Bug Description

With the istioctl version is 1.13.2 the following command showing error but when the istioctl version is 1.12.2 it is working fine:
`Error: failed to fetch mesh config: could not read configmap "istio" from namespace "istio-system": configmaps "istio" not found`

```
$ istioctl x describe pod sample-pod

Pod: sample-pod
   Pod Ports: 8080 (sample-container), 15090 (istio-proxy)
--------------------
Service: svc-sample
   Port: http 8080/HTTP targets pod port 8080
DestinationRule: drl-sample for "svc-sample.ns-sample.svc.cluster.local"
   Matching subsets: v1
   Traffic Policy TLS Mode: ISTIO_MUTUAL
   connection pool/outlier detection
VirtualService: vs-sample
   Weight 100%
   /*
RBAC policies: ns[ns-sample]-policy[aut-sample-sa-sample-allow]-rule[0], ns[ns-sample]-policy[aut-sample-sa-sample-allow]-rule[0], ns[ns-sample]-policy[aut-ns-sample-denyin]-rule[0]
--------------------
Error: failed to fetch mesh config: could not read configmap "istio" from namespace "istio-system": configmaps "istio" not found
```
With 1.12.2, it's working fine:
```
$ istioctl x describe pod sample-pod

Pod: sample-pod
   Pod Ports: 8080 (sample-container), 15090 (istio-proxy)
--------------------
Service: svc-sample
   Port: http 8080/HTTP targets pod port 8080
DestinationRule: drl-sample for "svc-sample.ns-sample.svc.cluster.local"
   Matching subsets: v1
   Traffic Policy TLS Mode: ISTIO_MUTUAL
   connection pool/outlier detection
VirtualService: vs-sample
   Weight 100%
   /*
RBAC policies: ns[ns-sample]-policy[aut-sample-sa-sample-allow]-rule[0], ns[ns-sample]-policy[aut-sample-sa-sample-allow]-rule[0], ns[ns-sample]-policy[aut-ns-sample-denyin]-rule[0]
Exposed on Ingress Gateway http://x.x.x.x
VirtualService: vs-sample
   Weight 100%
   /*
```
As the installation is dun with revision number so the configmap is writely named as istio-1-13-2-distroless.

```
$ kubectl get configmap -n istio-system

NAME                                       DATA   AGE
istio-1-13-2-distroless                    2      43d
istio-ca-root-cert                         1      376d
istio-gateway-deployment-leader            0      43d
istio-gateway-leader                       0      212d
istio-gateway-status-leader                0      43d
istio-leader                               0      376d
istio-namespace-controller-election        0      376d
istio-sidecar-injector-1-12-2-distroless   2      43d
istio-validation-controller-election       0      376d
kube-root-ca.crt                           1      296d

```

### Version

```prose
$ istioctl version
client version: 1.13.2
control plane version: 1.12.2
data plane version: 1.12.2 (2 proxies)

$ istioctl version
client version: 1.12.2
control plane version: 1.12.2
data plane version: 1.12.2 (2 proxies)

$ istioctl version
client version: 1.13.2
control plane version: 1.13.2
data plane version: 1.13.2 (2 proxies)

$ istioctl version
client version: 1.12.2
control plane version: 1.13.2
data plane version: 1.13.2 (2 proxies)

$ kubectl version --short
Client Version: v1.21.0
Server Version: v1.22.6
```

### Affected product area

- [ ] Docs
- [X] Installation
- [ ] Networking
- [ ] Performance and Scalability
- [ ] Extensions and Telemetry
- [ ] Security
- [X] Test and Release
- [ ] User Experience
- [X] Developer Infrastructure
- [X] Upgrade
- [ ] Multi Cluster
- [ ] Virtual Machine
- [ ] Control Plane Revisions

### Is this the right place to submit this?

- [X] This is not a security vulnerability
- [X] This is not a question about how to use Istio

**Comment by zirain** [comment]
can you share more details about the installation?

**Comment by soumyaparui** [comment]
Since istio 1.9.2 I have not changed the installation process. And it make use of istio operator (istio-operator.yaml) & revision tag
`istioctl install -f istio-operator.yaml --set revision=1-13-2-distroless`

**Comment by zirain** [comment]
I got it, you never install a default version istio?

**Comment by soumyaparui** [comment]
Hi,

No. Always like this, never used default version.

Same cluster, when I just switched back to istioctl version 1.12.X everything looks fine. Only issues are observed while using version 1.13.x.

- Soumya

**Comment by zirain** [comment]
> Hi, No. Always like this, never used default version. Same cluster, when I just switched back to istioctl version 1.12.X everything looks fine. Only issues are observed while using version 1.13.x. - Soumya
> […](#)
> On 28 Apr 2022, at 6:28 PM, zirain ***@***.***> wrote: ﻿ I got it, you never install a default version istio? — Reply to this email directly, view it on GitHub, or unsubscribe. You are receiving this because you authored the thread.

It's seems related to some new logic introduced with Istio v1.13.

**Comment by soumyaparui** [comment]
@zirain 
Please do note that I have opened one more issue for istioctl version 1.13.2 where looks like it is showing some invalid errors.

https://github.com/istio/istio/issues/38148

**Comment by soumyaparui** [comment]
> > Hi, No. Always like this, never used default version. Same cluster, when I just switched back to istioctl version 1.12.X everything looks fine. Only issues are observed while using version 1.13.x. - Soumya
> > […](#)
> > On 28 Apr 2022, at 6:28 PM, zirain _**@**_.***> wrote: ﻿ I got it, you never install a default version istio? — Reply to this email directly, view it on GitHub, or unsubscribe. You are receiving this because you authored the thread.
> 
> It seems related to some new logic introduced with Istio v1.13.

Yes seems like that. An until I get a definite answer I can not let our production clusters to upgrade to the latest version of istio. And rollout takes almost 6weeks as we have multiple clusters for different market.

**Comment by zirain** [comment]
root cause is here, `istioctl x describe pod` do not read the right revision.
https://github.com/istio/istio/blob/b43725b9d7fe4a71fcadbc36c743bc0d6b518107/istioctl/cmd/describe.go#L1205

related to PR: https://github.com/istio/istio/pull/36062

**Comment by soumyaparui** [comment]
Thanks for your quick response. Can you please have look at the other issue I opened and pending for sometime now. Because of these our production rollout of latest istio is on hold.

https://github.com/istio/istio/issues/38148

**Comment by soumyaparui** [comment]
@zirain 
Thanks for your response. So what is your suggestion, should we hold on until next patch release where this issue will be fixed or we can go ahead with rolling out the same in our cluster considering this as a know bug which won't impact actual functionality.

**Comment by zirain** [comment]
without the next patch, istioctl describe will not analyze the `PeerAuthentication`.

**Comment by soumyaparui** [comment]
@zirain Any idea when the patch will be available?

**Comment by zirain** [comment]
> @zirain Any idea when the patch will be available?

first we need get the PR merged, and wait for v1.13.4 released. @istio/release-managers-1-13 do we have a schedule for `v1.13.4`.

**Comment by soumyaparui** [comment]
@zirain Any idea when this will be release. We are waiting for this before production roll-out. If it is too late we may miss the EOL of 1.12.x which is ~June 2022 (Expected).
